bootstrap/kubeadm/api/v1beta1/kubeadm_types.go

```diff
@@ -244,6 +244,15 @@ type NodeRegistrationOptions struct {
 	// IgnorePreflightErrors provides a slice of pre-flight errors to be ignored when the current node is registered.
 	// +optional
 	IgnorePreflightErrors []string `json:"ignorePreflightErrors,omitempty"`
+
+	// ImagePullPolicy specifies the policy for image pulling
+	// during kubeadm "init" and "join" operations. The value of
+	// this field must be one of "Always", "IfNotPresent" or
+	// "Never". Defaults to "IfNotPresent".
+	// +kubebuilder:validation:Enum=Always;IfNotPresent;Never
+	// +kubebuilder:default=IfNotPresent
+	// +optional
+	ImagePullPolicy string `json:"imagePullPolicy,omitempty"`
```

Comment:
You also have to add the field to: bootstrap/kubeadm/types/upstreamv1beta3/types.go

Otherwise it won't be propagated to kubeadm.

But this doesn't explain your current issue